File: 장세훈/LeetCode/easy/InvertBinaryTree.java
class InvertBinaryTree {

    /*
        226. Invert Binary Tree
        https://leetcode.com/problems/invert-binary-tree/description/

        트리를 반전 시키는 문제
        dfs로 현재 노드의 자식 노드들을 스왑한다.

     */

    public TreeNode invertTree(TreeNode root) {
        dfs(root);
        return root;
    }

    void dfs(TreeNode node) {
        if (node == null) return;

        dfs(node.left);
        dfs(node.right);

        TreeNode temp = node.left;
        node.left = node.right;
        node.right = temp;
    }
}

File: 정수빈/LeetCode/easy/invert-binary-tree.py
# https://leetcode.com/problems/invert-binary-tree/?envType=study-plan-v2&envId=top-interview-150
# 설명: binary tree 기초 

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        
        def invert(node):
            if node == None:
                return 
            
            node_left = node.left
            node.left = node.right
            node.right = node_left

            invert(node.left)
            invert(node.right)

        invert(root)
        return root